### ISSUE TITLE
Make nice-looking plots of dependency trees

### ISSUE THREAD

**Issue by wlandau-lilly** [wlandau-lilly/drake#14]
It should be as simple as `DiagrammeR::from_igraph(build_graph(...))`, but [`from_igraph()` does not yet work for me](http://stackoverflow.com/questions/42985152/problems-converting-igraph-to-diagrammer-graph).

With `DiagrammeR`, I should also be able to assign different node colors for objects, functions, files, targets, etc. Maybe I can color edges by whether a target is created with a command or imported.

**Comment by wlandau-lilly** [comment]
See rich-iannone/DiagrammeR#219.

**Comment by wlandau-lilly** [comment]
@enantz-lilly suggested [visNetwork](http://datastorm-open.github.io/visNetwork). (This need not replace `DiagrammeR`. Users could choose the backend.)

```r
library(visNetwork)
igraph_object = drake:::dbug()$graph
network_data = toVisNetworkData(igraph_object)
visNetwork(nodes = network_data$nodes, edges = network_data$edges, height = "500px")
```

At minimum, I need to add directed edges. It would also be nice to have different-looking nodes for imports vs targets or files vs functions vs non-function objects.

**Comment by wlandau-lilly** [comment]
[CodeDepends](https://CRAN.R-project.org/package=CodeDepends) uses [Rgraphviz](http://www.bioconductor.org/packages/release/bioc/html/Rgraphviz.html), but that one is on [Bioconductor](http://www.bioconductor.org).

**Comment by wlandau-lilly** [comment]
I will release improved plotting functionality with the next version of drake. Here's what it looks like in the development version. Now is a great time for suggestions and requests. 

```r
attach(drake:::dbug()) # Workflow and environment I use for debugging
envir$f = function(x) g(x) + a + NOT_FOUND # Some symbols cannot be imported as objects.
plot_graph(plan = plan, envir = envir) # Build and render the graph
```

![graph](https://user-images.githubusercontent.com/22958003/28126843-b199cfd8-66f8-11e7-9859-b2872db67c31.PNG)

**Comment by wlandau-lilly** [comment]
Even better: let's sort the graph from left to right and show everyone how the parallel computing works. In the new graph below, each column is a separate parallelizable stage. When you call `make()`, you go from left to right, and the targets in the current column are distributed over the available parallel jobs. In other words, you have parallelism within columns, and you move sequentially from one column to the next.  In the example below, we can do the first 7 targets in parallel, then `j` and `myinput` in parallel, and then there's the unavoidably sequential move to `h`, and so on.

![better](https://user-images.githubusercontent.com/22958003/28134495-1d480caa-6711-11e7-8070-4d8ad4be1675.PNG)

For those who are curious, I am now happily committed to [visNetwork](https://CRAN.R-project.org/package=visNetwork). I set the "level" column in the "nodes" data frame in `visNetwork()` and then call `visHierarchicalLayout()`.
